bulk/text.py

```diff
@@ -1,46 +1,54 @@
-from tornado.ioloop import IOLoop
-
 import numpy as np
 import pandas as pd
-
-from bokeh.io import curdoc
 from bokeh.layouts import column, row
-from bokeh.models import (Button, ColumnDataSource, DataTable, TableColumn, TextInput)
+from bokeh.models import Button, ColumnDataSource, TextInput, DataTable, TableColumn, ColorBar
 from bokeh.plotting import figure
-from bokeh.models import DataTable, TableColumn
 
-def bulk_text(path):
+from .utils import get_color_mapping, get_datatable_columns
 
+
+def bulk_text(path):
     def bkapp(doc):
         df = pd.read_csv(path)
-        saved_idx = []
-        highlighed_idx = []
+        highlighted_idx = []
 
-        source = ColumnDataSource(data=dict())
-        source_orig = ColumnDataSource(data=df)
+        mapper, df = get_color_mapping(df)
+        datatable_columns = get_datatable_columns(df)
+        columns = [
+            TableColumn(field=col, title=col) for col in datatable_columns
+        ]
 
         def update(attr, old, new):
-            global highlighed_idx
+            """Callback used for plot update when lasso selecting"""
+            global highlighted_idx
             subset = df.iloc[new]
-            highlighed_idx = new
+            highlighted_idx = new
             subset = subset.iloc[np.random.permutation(len(subset))]
             source.data = subset
 
         def save():
+            """Callback used to save highlighted data points"""
             global highlighed_idx
-            df.iloc[highlighed_idx][['text']].to_csv(text_filename.value)
+            df.iloc[highlighted_idx].to_csv(text_filename.value)
 
-        columns = [
-            TableColumn(field="text", title="text")
-        ]
+        source = ColumnDataSource(data=dict())
+        source_orig = ColumnDataSource(data=df)
 
         data_table = DataTable(source=source, columns=columns, width=800)
         source.data = df
 
         p = figure(title="", sizing_mode="scale_both", tools="lasso_select")
-        scatter = p.circle(x="x", y="y", size=1, source=source_orig)
-        p.plot_width = 300
-        p.plot_height = 300
+
+        circle_kwargs = {"x": "x", "y": "y", "size": 1, "source": source_orig}
+        if "color" in df.columns:
+            circle_kwargs.update({"color": mapper})
+
+            color_bar = ColorBar(color_mapper=mapper['transform'], width=8)
+            p.add_layout(color_bar, 'right')
+
+        scatter = p.circle(**circle_kwargs)
+        p.plot_width = 600
+        p.plot_height = 600
```

Comment:
Ah! So that's why the points feel much smaller than before. 

While I wouldn't mind giving the user some way to change the styles of the chart (maybe via a config file of sorts?) can we leave the base settings as-is for now?

Reply:
Another thing about this: I'll likely add a zoom widget to the chart after this PR is merged. That way you can zoom in properly if you need more space to draw the shape.

Reply:
I think that the zooming feature is really relevant, I found myself struggling a bit by drawing the shape sometimes..

Reply:
If you wanna open an issue, I can have a look at it 😃

Reply:
It's just an [extra widget](https://docs.bokeh.org/en/latest/docs/user_guide/tools.html#scroll-pinch-tools) that needs to be configured for Bokeh. 

But feel free to make a PR for it 👍

Reply:
Will do !